include/pybind11/stl.h

```diff
@@ -43,17 +43,15 @@ template <typename Value, typename Alloc> struct type_caster<std::vector<Value,
     }
 
     static PyObject *cast(const type &src, return_value_policy policy, PyObject *parent) {
-        PyObject *list = PyList_New(src.size());
+        object list(PyList_New(src.size()), false);
```

Comment:
A followup comment on this:

I think we should use the pybind11 [list](https://github.com/tmiasko/pybind11/blob/e2242538ed8a43011668095406c451dd0381b745/include/pybind11/pytypes.h#L396-L403) class for this. That way we get a more simple syntax, and checking for `NULL` can be deferred to the list class, where it should (IMO) be done.